src/pip/_internal/req/req_install.py

```diff
@@ -150,6 +150,8 @@ def __init__(
         self.global_options = global_options if global_options else []
         self.hash_options = hash_options if hash_options else {}
         self.config_settings = config_settings
+        if isinstance(comes_from, InstallRequirement) and comes_from.config_settings:
```

Comment:
should it provide config-settings also for the package dependencies?

Reply:
Not sure I understand the question; could you clarify?

Reply:
In the `test_config_settings_local_to_package` the package simple1 depends on the package bar. Should bar_args also contain --build-option, like simple1_args does?

Reply:
Ah I see, build flag propagation. We currently do propagate setuptools build options, so I think `--config-settings` should do the same (unless the dependency explicitly specifies its own `--config-settings`). There are people who don’t like this (see #1883) but I think we should stick with the current behaviour, as mentioned in the thread.